awswrangler/distributed/ray/s3/_list.py

```diff
@@ -0,0 +1,67 @@
+"""Ray S3 List module (PRIVATE)."""
+import datetime
+import fnmatch
+import logging
+from typing import TYPE_CHECKING, Any, Dict, Iterator, List, Optional, Union
+
+from pyarrow.fs import FileSelector, FileType, _resolve_filesystem_and_path
+
+if TYPE_CHECKING:
+    from mypy_boto3_s3 import S3Client
+
+_logger: logging.Logger = logging.getLogger(__name__)
+
+
+def _list_objects_s3fs(  # pylint: disable=unused-argument
```

Comment:
Because this is a generator, and the non-distributed alternative returns a list, and the rest of the code expects a list it seems, do we need to evaluate the generator with `list()`? I'm hoping not but just to confirm

Reply:
The non-distributed version returns a generator as well